cmake/Dependencies.cmake

```diff
@@ -0,0 +1,247 @@
+  PATH_SUFFIXES Debug
+)
+add_library(sasl2 STATIC IMPORTED)
+set_target_properties(sasl2 PROPERTIES IMPORTED_LOCATION_DEBUG "${SASL2_LIB_DEBUG}")
+set_target_properties(sasl2 PROPERTIES IMPORTED_LOCATION_RELEASE "${SASL2_LIB_RELEASE}")
+
+find_library(
+  HTTP_PARSER_LIB_RELEASE
+  NAMES http_parser
+  PATHS ${MESOS_ROOT}/build/3rdparty/http_parser-${HTTP_PARSER_VERSION}/src/http_parser-${HTTP_PARSER_VERSION}-build
+  PATH_SUFFIXES Release
+)
+find_library(
+  HTTP_PARSER_LIB_DEBUG
+  NAMES http_parser
+  PATHS ${MESOS_ROOT}/build/3rdparty/http_parser-${HTTP_PARSER_VERSION}/src/http_parser-${HTTP_PARSER_VERSION}-build
+  PATH_SUFFIXES Debug
+)
+add_library(http_parser STATIC IMPORTED)
+set_target_properties(http_parser PROPERTIES IMPORTED_LOCATION_DEBUG "${HTTP_PARSER_LIB_DEBUG}")
+set_target_properties(http_parser PROPERTIES IMPORTED_LOCATION_RELEASE "${HTTP_PARSER_LIB_RELEASE}")
+
+find_library(
+  ZOOKEEPER_LIB_RELEASE
+  NAMES zookeeper
+  PATHS ${MESOS_ROOT}/build/3rdparty/zookeeper-${ZOOKEEPER_VERSION}/src/zookeeper-${ZOOKEEPER_VERSION}-build
+  PATH_SUFFIXES Release
+)
+find_library(
+  ZOOKEEPER_LIB_DEBUG
+  NAMES zookeeper
+  PATHS ${MESOS_ROOT}/build/3rdparty/zookeeper-${ZOOKEEPER_VERSION}/src/zookeeper-${ZOOKEEPER_VERSION}-build
+  PATH_SUFFIXES Debug
+)
+add_library(zookeeper STATIC IMPORTED)
+set_target_properties(zookeeper PROPERTIES IMPORTED_LOCATION_DEBUG "${ZOOKEEPER_LIB_DEBUG}")
+set_target_properties(zookeeper PROPERTIES IMPORTED_LOCATION_RELEASE "${ZOOKEEPER_LIB_RELEASE}")
+
+find_library(
+  ZOOKEEPER_HASHTABLE_LIB_RELEASE
+  NAMES hashtable
+  PATHS ${MESOS_ROOT}/build/3rdparty/zookeeper-${ZOOKEEPER_VERSION}/src/zookeeper-${ZOOKEEPER_VERSION}-build
+  PATH_SUFFIXES Release
+)
+find_library(
+  ZOOKEEPER_HASHTABLE_LIB_DEBUG
+  NAMES hashtable
+  PATHS ${MESOS_ROOT}/build/3rdparty/zookeeper-${ZOOKEEPER_VERSION}/src/zookeeper-${ZOOKEEPER_VERSION}-build
+  PATH_SUFFIXES Debug
+)
+add_library(zookeeper_hashtable STATIC IMPORTED)
+set_target_properties(zookeeper_hashtable PROPERTIES IMPORTED_LOCATION_DEBUG "${ZOOKEEPER_HASHTABLE_LIB_DEBUG}")
+set_target_properties(zookeeper_hashtable PROPERTIES IMPORTED_LOCATION_RELEASE "${ZOOKEEPER_HASHTABLE_LIB_RELEASE}")
+
+find_library(
+  MESOS_PROTOBUFS_LIB_RELEASE
+  NAMES mesos-protobufs
+  PATHS ${MESOS_ROOT}/build/src
+  PATH_SUFFIXES Release
+)
+find_library(
+  MESOS_PROTOBUFS_LIB_DEBUG
+  NAMES mesos-protobufs
+  PATHS ${MESOS_ROOT}/build/src
+  PATH_SUFFIXES Debug
+)
+add_library(mesos_protobufs STATIC IMPORTED)
+set_target_properties(mesos_protobufs PROPERTIES IMPORTED_LOCATION_DEBUG "${MESOS_PROTOBUFS_LIB_DEBUG}")
+set_target_properties(mesos_protobufs PROPERTIES IMPORTED_LOCATION_RELEASE "${MESOS_PROTOBUFS_LIB_RELEASE}")
+
+find_library(
+  MESOS_LIB_RELEASE
+  NAMES mesos
+  PATHS ${MESOS_ROOT}/build/src
+  PATH_SUFFIXES Release
+)
+find_library(
+  MESOS_LIB_DEBUG
+  NAMES mesos
+  PATHS ${MESOS_ROOT}/build/src
+  PATH_SUFFIXES Debug
+)
+add_library(mesos STATIC IMPORTED)
```

Comment:
Do I understand correctly that this statically links Mesos into each dll? 
If yes, then what is the resulting size of each dll?

Reply:
Yes.

libjournaldlogger.dll 5.0M
logsink.dll 267K
metrics.dll 5.3M